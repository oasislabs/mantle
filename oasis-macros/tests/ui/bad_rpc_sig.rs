oasis_std::contract! {

#[derive(Contract)]
pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: &Context) -> Self {
        Self(42)
    }

    pub unsafe extern "C" fn wtf<T: std::fmt::Debug>(&self, ctx: &Context, val: T) {
        println!("val: {:?}", val);
    }
}

}

fn main() {}
