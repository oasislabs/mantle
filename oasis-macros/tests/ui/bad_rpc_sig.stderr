error: RPC methods cannot declare an ABI.
  --> $DIR/bad_rpc_sig.rs:11:16
   |
11 |     pub unsafe extern "C" fn wtf<T: std::fmt::Debug>(&self, ctx: &Context, val: T) -> Result<()> {
   |                ^^^^^^

error: RPC methods may not be unsafe.
  --> $DIR/bad_rpc_sig.rs:11:9
   |
11 |     pub unsafe extern "C" fn wtf<T: std::fmt::Debug>(&self, ctx: &Context, val: T) -> Result<()> {
   |         ^^^^^^

error: RPC methods may not have generic type parameters.
  --> $DIR/bad_rpc_sig.rs:11:33
   |
11 |     pub unsafe extern "C" fn wtf<T: std::fmt::Debug>(&self, ctx: &Context, val: T) -> Result<()> {
   |                                 ^

error: aborting due to 3 previous errors

